experimental/cuda2/cuda_dynamic_symbols.c

```diff
@@ -20,21 +20,25 @@ static const char* iree_hal_cuda_dylib_names[] = {
 #endif  // IREE_PLATFORM_WINDOWS
 };
 
-#define IREE_CONCAT(A, B) A B
+// CUDA API version for cuGetProcAddress.
+// 1000 * major + 10 * minor
+#define IREE_CUDA_DRIVER_API_VERSION 11030
 
-// Resolves all CUDA dynamic symbols in `dynamic_symbol_tables.h`, prefer _v2
-// version if it exists.
-static iree_status_t iree_hal_cuda2_dynamic_symbols_resolve_all(
-    iree_hal_cuda2_dynamic_symbols_t* syms) {
-#define IREE_CU_PFN_DECL(cuda_symbol_name, ...)                         \
-  {                                                                     \
-    static const char* name = #cuda_symbol_name;                        \
-    IREE_RETURN_IF_ERROR(iree_dynamic_library_lookup_symbol(            \
-        syms->dylib, name, (void**)&syms->cuda_symbol_name));           \
-    static const char* name_v2 = IREE_CONCAT(#cuda_symbol_name, "_v2"); \
-    void* fptr_v2;                                                      \
-    iree_dynamic_library_lookup_symbol(syms->dylib, name_v2, &fptr_v2); \
-    if (fptr_v2) syms->cuda_symbol_name = fptr_v2;                      \
+// Load CUDA entry points.
+static iree_status_t iree_hal_cuda_dynamic_symbols_resolve_all(
+    iree_hal_cuda_dynamic_symbols_t* syms) {
+  CUresult (*cuGetProcAddress)(const char*, void**, int, cuuint64_t);
```

Comment:
you can put this in the dynamic_symbol_tables.h and do the lookup into it (&syms->cuGetProcAddress) - then the code below can use the IREE_CUDA_RETURN_IF_ERROR macro that handles converting CUresults to statuses with nice messages

so something like `IREE_CUDA_RETURN_IF_ERROR(syms, cuGetProcAddress(...), "resolving " # cuda_symbol_name);`

Reply:
(advantage of this being that plugins/etc can get the symbol for querying their own symbols that we don't include in the base table)

Reply:
Sounds good, thanks! Should we move `cuGetErrorName`/`cuGetErrorString` to the top of the table so they are loaded first incase they are needed by `IREE_CUDA_RETURN_IF_ERROR` to decode any errors?

Reply:
It seems `CUDA_RETURN_IF_ERROR` is eating the `__VA_ARGS__` when used inside `CU_PFN_DECL` and causing some weird compilation errors. Any advice? @benvanik

Reply:
@benvanik I took a stab at it. Let me know if you think of a cleaner way to handle the error checking.

Reply:
Huh, hadn't considered that! What you have looks reasonable as I doubt those functions are going to change much/ever be missing.